src/utils/hooks/useAlerts/utils/utils.ts

```diff
@@ -0,0 +1,87 @@
+import { murmur3 } from 'murmurhash-js';
+
+import { isEmpty } from '@kubevirt-utils/utils/utils';
+import {
+  Alert,
+  AlertStates,
+  PrometheusRule,
+  Rule,
+  RuleStates,
+  Silence,
+  SilenceStates,
+} from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Group,
+  MONITORING_SALT,
+  PrometheusRulesResponse,
+} from '@virtualmachines/details/tabs/overview/utils/utils';
+
+export const addAlertIdToRule = (group: Group, rule: PrometheusRule): Rule => {
+  const key = [
+    group?.file,
+    group?.name,
+    rule?.name,
+    rule?.duration,
+    rule?.query,
+    ...Object?.entries(rule?.labels).map(([k, v]) => `${v}=${k}`),
+  ].join(',');
+  return { ...rule, id: String(murmur3(key, MONITORING_SALT)) };
+};
+
+export const getAlertsAndRules = (
+  data: PrometheusRulesResponse['data'],
+): { alerts: Alert[]; rules: Rule[] } => {
+  // Flatten the rules data to make it easier to work with, discard non-alerting rules since those
+  // are the only ones we will be using and add a unique ID to each rule.
+  const groups = data?.groups as PrometheusRulesResponse['data']['groups'];
+  const rules = groups?.flatMap((group) =>
+    group?.rules
+      ?.filter((rule) => rule?.type === 'alerting')
+      ?.map((rule) => addAlertIdToRule(group, rule)),
+  );
+
+  // Add rule object to each alert
+  const alerts = rules?.flatMap((rule) => rule?.alerts?.map((alert) => ({ rule, ...alert })));
+
+  return { alerts, rules };
+};
+
+export const isSilenced = (alert: Alert, silence: Silence): boolean =>
+  [AlertStates.Firing, AlertStates.Silenced].includes(alert.state) &&
+  silence?.matchers?.every((m) => {
```

Comment:
Can you please add a more meaningful name for m :) ?

Reply:
Good catch. I had to fix a lot of those. I should have known I was going to miss one. =)